doc/autogen/types/bytes.rst

```diff
@@ -36,7 +36,8 @@
 
     Matches the ``bytes`` object against the regular expression *regex*.
     Returns the matching part or, if *group* is given, then the
-    corresponding subgroup.
+    corresponding subgroup. The expression is considered anchored to the
```

Comment:
I am sure that every individual works out their mental model for comprehending the difference between match() and find(), in a manner such that there is no one way of expressing it perfectly for all audiences. But the term `anchored` that you use, could be combined with mention of the `^` regexp operator, then to likely  achieve wider correct comprehension for people whose conception in built in terms of that `^` regexp operator. So I suggest something like: `Returns the matching part or, if *group* is given, then the corresponding subgroup. The expression is considered anchored, as though it starts with an implicit ^ regexp operator, to the beginning of the data.`